src/Features/LanguageServer/ProtocolUnitTests/VSTypeScriptHandlerTests.cs

```diff
@@ -80,6 +81,22 @@ public async Task TestRoslynTypeScriptHandlerInvoked()
         Assert.Empty(response);
     }
 
+    [Fact, WorkItem("https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1901118")]
+    public async Task TestGetSimplifierOptionsOnTypeScriptDocument()
+    {
+        var workspaceXml =
+@$"<Workspace>
+    <Project Language=""TypeScript"" CommonReferences=""true"" AssemblyName=""TypeScriptProj"">
+        <Document FilePath=""C:\T.ts""></Document>
+    </Project>
+</Workspace>";
+
+        await using var testLspServer = await CreateTsTestLspServerAsync(workspaceXml);
+        var document = testLspServer.GetCurrentSolution().Projects.Single().Documents.Single();
+        var simplifierOptions = testLspServer.TestWorkspace.GlobalOptions.GetSimplifierOptions(document.Project.Services, fallbackOptions: null);
```

Comment:
What is calling this in the first case though on TypeScript projects? Should that code not be running in the first place?

Reply:
I added this to the bug, but since it's not public I'll add it here too. Stack at the time of the exception:

```
   at Microsoft.CodeAnalysis.Host.HostLanguageServices.GetRequiredService[TLanguageService]()
   at Microsoft.CodeAnalysis.Diagnostics.EngineV2.DiagnosticIncrementalAnalyzer.ProjectState.<MergeAsync>d__16.MoveNext() in /_/src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs:line 270
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in System.Runtime.ExceptionServices\ExceptionDispatchInfo.cs:line 72
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) in System.Runtime.CompilerServices\TaskAwaiter.cs:line 88
   at Microsoft.CodeAnalysis.Diagnostics.EngineV2.DiagnosticIncrementalAnalyzer.StateSet.<OnDocumentClosedAsync>d__14.MoveNext() in /_/src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateSet.cs:line 140
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in System.Runtime.ExceptionServices\ExceptionDispatchInfo.cs:line 72
   at Microsoft.CodeAnalysis.Diagnostics.EngineV2.DiagnosticIncrementalAnalyzer.<TextDocumentCloseAsync>d__99.MoveNext() in /_/src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs:line 221
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in System.Runtime.ExceptionServices\ExceptionDispatchInfo.cs:line 72
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) in System.Runtime.CompilerServices\TaskAwaiter.cs:line 88
   at Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.NormalPriorityProcessor.<<ProcessCloseDocumentIfNeededAsync>g__DocumentCloseAsync|28_0>d.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in System.Runtime.ExceptionServices\ExceptionDispatchInfo.cs:line 72
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) in System.Runtime.CompilerServices\TaskAwaiter.cs:line 88
   at Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.<>c__DisplayClass25_1`1.<<RunAnalyzersAsync>b__0>d.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in System.Runtime.ExceptionServices\ExceptionDispatchInfo.cs:line 72
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) in System.Runtime.CompilerServices\TaskAwaiter.cs:line 88
   at Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.<GetOrDefaultAsync>d__27`2.MoveNext()
```

Reply:
Yes, @sharwell is correct. It is being called here: https://github.com/dotnet/roslyn/blob/85de7fcfbc284f882a14d0cffeb41875608ff235/src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs#L270

We can add defensive checks on the callsites, but that doesn't prevent something similar to pop up in the future, and the change to this method seemed a reasonable one - any language that doesn't support simplification will most likely ignore the value for the SimplifierOptions, so the CommonDefaults should work just fine.

Reply:
Oh totally my bad: I missed that and though we were missing a few frames.